llama-index-integrations/readers/llama-index-readers-toggl/llama_index/readers/toggl/base.py

```diff
@@ -0,0 +1,81 @@
+import asyncio
+import datetime
+import json
+from typing import List, Optional
+
+from llama_index.core.readers.base import BaseReader
+from llama_index.core.schema import Document
+
+from llama_index.readers.toggl.dto import TogglTrackItem, TogglOutFormat
+
+
+class TogglReader(BaseReader):
+    def __init__(
+        self, api_token: str, user_agent: str = "llama_index_toggl_reader"
+    ) -> None:
+        """Initialize with parameters."""
+        super().__init__()
+        self.api_token = api_token
+        self.user_agent = user_agent
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
+
+    def load_data(
+        self,
+        workspace_id: str,
+        project_id: str,
+        start_date: Optional[datetime.datetime] = None,
+        end_date: Optional[datetime.datetime] = datetime.datetime.now(),
+        out_format: TogglOutFormat = TogglOutFormat.json,
+    ) -> List[Document]:
+        """Load data from Toggl.
+
+        Args:
+            workspace_id (str): The workspace ID.
+            project_id (str): The project ID.
+            start_date (Optional[datetime.datetime]): The start date.
+            end_date (Optional[datetime.datetime]): The end date.
+            out_format (TogglOutFormat): The output format.
+        """
+        return self.loop.run_until_complete(
+            self._load_data(workspace_id, project_id, start_date, end_date, out_format)
+        )
+
+    async def _load_data(
```

Comment:
the base class actually has a `aload_data()` method -- we should implement that for async

Reply:
fixed, I moved `_load_data` to `aload_data` and referenced it in `load_data`

Reply:
nice!